tests/test_argument.py

```diff
@@ -75,6 +76,36 @@ def test_symbol_add(self):
         new = x + y
         self.assertIsInstance(new, sympy.Add)
 
+    def test_pickle(self):
+        """
+        Make sure attributes are preserved when pickling
+        """
+        A = Parameter('A', min=0., max=1e3, fixed=True)
+        new_A = pickle.loads(pickle.dumps(A))
```

Comment:
Perhaps you can add the same test for automatically generated names for as much as possible?

Reply:
Do we want to test against deprecated behaviour?

Reply:
It's depricated, but still one of the futures now right? So it should be pickalable, no?

Reply:
Whatever floats your boat ;)